gauge/specification.go

```diff
@@ -307,13 +307,15 @@ func (spec *Specification) Filter(filter SpecItemFilter) (*Specification, *Speci
 	specWithOtherItems := new(Specification)
 	*specWithFilteredItems, *specWithOtherItems = *spec, *spec
 	for i := 0; i < len(specWithFilteredItems.Items); i++ {
-		if filter.Filter(specWithFilteredItems.Items[i]) {
+		item := specWithFilteredItems.Items[i]
+		if item.Kind() == ScenarioKind && filter.Filter(item) {
```

Comment:
Do we need the check for `item.Kind() == ScenarioKind` - I believe all the three implementation of `SpecItemFilter` have this check inside:

- https://github.com/getgauge/gauge/blob/4df79cc735e766f2fef50fd3bc90e9e5cf4a9445/filter/specItemFilter.go#L51
- https://github.com/getgauge/gauge/blob/4df79cc735e766f2fef50fd3bc90e9e5cf4a9445/filter/specItemFilter.go#L67
- https://github.com/getgauge/gauge/blob/4df79cc735e766f2fef50fd3bc90e9e5cf4a9445/filter/specItemFilter.go#L82

Reply:
Yes, we need to do that as without this we will endup removing non `ScenarioKind` items either from `specWithFilteredItems` or `specWithOtherItems`. 
It was working perior to parallel tag filter fature. As we were not using the filtered  out specs.

Reply:
So probably we can move that check out of those `Filter` function to the function whcih uses them.

Reply:
yeah, because right now it looks like we are checking in multiple locations.